README.md

```diff
@@ -9,6 +9,50 @@
 npm install screwdriver-build-bookend
 ```
 
+### Using the bookend interface
+Extend the bookend interface included in this module. You will need to define `getSetupCommand` and `getTeardownCommand` to return the commands needed to execute and return these in a promise.
+
+```js
+const { BookendInterface } = require('screwdriver-build-bookend');
+
+class MyBookend extends BookendInterface {
+    getSetupCommand() {
+        return Promise.resolve('echo "hello world"');
+    }
+    getTeardownCommand() {
+        return Promise.resolve('echo "goodbye world"');
+    }
+}
+
+module.exports = MyBookend;
+```
+
+### Getting final bookend commands
+Use the Bookend module to combine a set of BookendInterface based modules into single set of setup and teardown commands. See more examples in [the tests](https://github.com/screwdriver-cd/screwdriver-build-bookend/blob/master/test/index.test.js).
+
+```js
+const SampleBookend = require('sd-sample-bookend');
+const { Bookend } = require('screwdriver-build-bookend');
+const b = new Bookend(
+    // Provide a set of default instantiated plugins
+    { 'sd-sample': new SampleBookend() },
```

Comment:
Can you add comment for the type of `defaultModules` in the function definition? 
https://github.com/screwdriver-cd/build-bookend/blob/master/index.js#L30-L36